### PR TITLE
[PoC] Hits Content Plugin

### PR DESCRIPTION
**Pull request by andrepereiradasilva**
### Summary of Changes

Working PoC for a new Hits Content Plugin That controls the hits through the components.

Also fixes some advantages:
- Now for contacts/newsfeeds will also hit categories. And also for all other components with categories (ex: banners, weblinks).
- performance improvement since the hit is now generated after the page is generated by php (one less costy update sql in all article, category views pages)
- Since it's a plugin You can disable the hits if not needed!
### Testing Instructions
1. Code review
2. [TO DO]
### Documentation Changes Required

None.
### Notes

This is not complete work.

The part that i really have doubts is the part where we process the ajax call, ie, where we make the hit itself.

In this PoC i searched for the model based contenxt and if from onContentBeforeDisplay event, then loaded the model and check if the model has the `hit` method and execute it. But is this the best way to do it?

### CODE REVIEW

File: plugins/content/hit/hit.php [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+		// Check if model is loaded and _context property exists.
+		if (!$model || $model->get('_context') !== $context)
+		{
+			// Do nothing.
+			echo new JResponseJson();
+			$this->app->close();
+		}
+
+		// Set the hit counter
+		$hitcount = $this->app->input->set('hitcount', 1);
```
Comment (csthomas) [inline plugins/content/hit/hit.php]: IMO line with `$hitcount = ...` is not needed.
On models we get 1 by default.

``` php
$input->getInt('hitcount', 1);
```

Comment (andrepereiradasilva) [reply plugins/content/hit/hit.php]: probably a leftover of messing around will check

Comment (andrepereiradasilva) [reply plugins/content/hit/hit.php]: ok removed

**Comment by andrepereiradasilva** [comment]
hum i think this can't be done in onContentBeforeDisplay because we need this also in categories and other types. will have to investigate further

this needs to be page related, not ocntent related

**Comment by andrepereiradasilva** [comment]
i now moved the hit behaviour to after dispatch.
in reality i think this now should be a system plugin, not a content plugin, but not sure, maybe can be needed to add hit somewhere in the contentprepare in the future, so i think i wwill leave it like this.

**Pull request closed without merging**